onnxruntime/core/session/inference_session.cc

```diff
@@ -1725,6 +1733,26 @@ common::Status InferenceSession::Run(const RunOptions& run_options, IOBinding& i
              &io_binding.GetOutputs(), &io_binding.GetOutputsDeviceInfo());
 }
 
+#ifdef ENABLE_TRAINING
+int64_t InferenceSession::CreatePartialRun() {
+  return session_state_->GetPartialGraphExecutionManager().GetNewPartialGraphRunId();
+}
+
+common::Status InferenceSession::PartialRun(IOBinding& io_binding, int64_t run_id) {
+  ORT_ENFORCE(session_state_->GetEnableMemoryPattern() == false);
+  ORT_ENFORCE(session_state_->GetEnableMemoryReuse() == false);
+  ORT_ENFORCE(session_state_->GetPartialGraphExecutionManager().IsValidRunId(run_id));
```

Comment:
Let's make this return a proper failed Status with error message, when run_id is not found.